Libraries/LibWeb/CredentialManagement/Credential.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2025, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Forward.h>
+#include <LibWeb/Bindings/CredentialPrototype.h>
+#include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/WebIDL/Promise.h>
+
+namespace Web::CredentialManagement {
+
+class Credential : public Bindings::PlatformObject {
+    WEB_PLATFORM_OBJECT(Credential, Bindings::PlatformObject);
+    GC_DECLARE_ALLOCATOR(Credential);
+
+public:
+    [[nodiscard]] static GC::Ref<Credential> create(JS::Realm&);
+
+    static JS::ThrowCompletionOr<GC::Ref<WebIDL::Promise>> is_conditional_mediation_available(JS::VM&);
```

Comment:
methods that return a promise should (?) almost always return just a `Ref<Promise>`, rather than a `ThrowCompletionOr<Promise>`. If there's an error, specs almost always say to return a promise rejected with xyz, or to reject p with xyz and then return p.

Reply:
Looks like that's the standard, the only one that does it differently is `SubtleCrypto` which was the one I took inspiration from.